internal/monitor/otelexporters.go

```diff
@@ -36,20 +40,57 @@ const serviceName = "gcsfuse"
 
 // SetupOTelMetricExporters sets up the metrics exporters
 func SetupOTelMetricExporters(ctx context.Context, c *cfg.Config) (shutdownFn common.ShutdownFn) {
+	shutdownFns := make([]common.ShutdownFn, 0)
 	options := make([]metric.Option, 0)
+
 	opts, shutdownFn := setupPrometheus(c.Metrics.PrometheusPort)
 	options = append(options, opts...)
+	shutdownFns = append(shutdownFns, shutdownFn)
+
+	opts, shutdownFn = setupCloudMonitoring(c.Metrics.CloudMetricsExportIntervalSecs)
+	options = append(options, opts...)
+	shutdownFns = append(shutdownFns, shutdownFn)
+
 	res, err := getResource(ctx)
 	if err != nil {
 		logger.Errorf("Error while fetching resource: %v", err)
 	} else {
 		options = append(options, metric.WithResource(res))
 	}
+
 	meterProvider := metric.NewMeterProvider(options...)
+	shutdownFns = append(shutdownFns, meterProvider.Shutdown)
+
 	otel.SetMeterProvider(meterProvider)
-	return common.JoinShutdownFunc(meterProvider.Shutdown, shutdownFn)
+
+	return common.JoinShutdownFunc(shutdownFns...)
 }
 
+func setupCloudMonitoring(secs int64) ([]metric.Option, common.ShutdownFn) {
+	if secs <= 0 {
+		return nil, nil
+	}
+	options := []cloudmetric.Option{
+		cloudmetric.WithMetricDescriptorTypeFormatter(metricFormatter),
+		cloudmetric.WithFilteredResourceAttributes(func(kv attribute.KeyValue) bool {
+			// Ensure that PID is available as a metric label on metrics explorer.
```

Comment:
Any more info regarding PID?

Reply:
Can you elaborate on what information you are looking for?

Reply:
Was trying to understand what is semconv.ProcessPIDKey and why we are setting this. Is this to differentiate different PIDs ( different mounts ) on the same node or is it for something else.
wanted to add metrics for the things which we can map to debugging and checking if we can reduce cardinality wherever possible.

Reply:
Yes, it's for differentiating between mounts.

Reply:
I feel this will break the cardinality. PIDs can be a long number ( maybe 5 digits). And filtering support for them needs means they have to be bucketized and not sure if metrics pipeline can take it. Generally only a fixed set of values is what can be filtered. Here we already have bucket name, pod name etc . Can we recheck this if metric pipeline can take this or not.

Reply:
This is a different exporter i.e. Cloud monitoring (aka stackdriver) - unrelated to Prometheus. FWIW, the corresponding exporter in OpenCensus already exported PID.

Reply:
We could take a separate task to audit the number of dimensions and cardinality for all metrics in the end of the feature. For now resolving this.